# Extracción de datos de la PROFECO para los precios del Aguacte Hass

In [1]:
import requests
import pandas as pd

# 1. Definimos la URL exacta que interceptaste
url = "https://qqp.profeco.gob.mx/api/precios?clave_ciudad=&busqueda=aguacate"

# 2. Replicamos los headers esenciales
headers = {
    "accept": "*/*",
    "accept-language": "es-MX,es-ES;q=0.9,es;q=0.8",
    "authorization": "Basic cXFwd2ViOnFxcHdlYg==",
    # Es buena práctica poner un User-Agent para simular ser un navegador
    "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
}

# 3. Hacemos la petición GET
print("Realizando la petición a la API de PROFECO...")
response = requests.get(url, headers=headers)

# 4. Verificamos que la petición fue exitosa (Código 200)
if response.status_code == 200:
    # Convertimos la respuesta a JSON
    data = response.json()

    # IMPORTANTE: Aquí dependemos de la estructura del JSON.
    # Usualmente los datos vienen en una llave como 'data', 'resultados', o directamente es una lista.
    # Si 'data' es una lista de diccionarios, Pandas lo convierte directo:

    # df_profeco = pd.DataFrame(data['resultados']) # Ajusta 'resultados' según lo que imprima el JSON

    # Por ahora, vamos a imprimir las llaves del JSON para ver cómo viene estructurado
    if isinstance(data, dict):
        print("Estructura del JSON (llaves):", data.keys())
    else:
        print("El JSON es una lista con", len(data), "registros.")
        df_profeco = pd.DataFrame(data)
        display(df_profeco.head())

else:
    print(f"Error al conectar: {response.status_code}")
    print(response.text)

Realizando la petición a la API de PROFECO...
Estructura del JSON (llaves): dict_keys(['success', 'message', 'data'])


In [7]:
#print(len(data['data']['productos'][0]))

print(data['data']['productos'][0])

{'producto': 'AGUACATE, S/M, 1 KG. HASS', 'tipo_producto': 'AGUACATE', 'precio': 59, 'fecha_observacion': '2026-04-29', 'cadena_comercial': 'WAL-MART EXPRESS', 'establecimiento': 'WALMART EXPRESS SUCURSAL GIRASOLES', 'direccion': 'RANCHO SAN LORENZO 6, ESQ. CALZADA DE LAS BOMBAS (INT.', 'colonia': 'GIRASOLES III', 'cp': '04920', 'municipio': 'Coyoacán', 'entidad': 'Ciudad de México'}


In [8]:
df_profeco = pd.DataFrame(data['data']['productos'])

df_profeco.head(10)

,producto,tipo_producto,precio,fecha_observacion,cadena_comercial,establecimiento,direccion,colonia,cp,municipio,entidad
0,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,59.0,2026-04-29,WAL-MART EXPRESS,WALMART EXPRESS SUCURSAL GIRASOLES,"RANCHO SAN LORENZO 6, ESQ. CALZADA DE LAS BOMB...",GIRASOLES III,04920,Coyoacán,Ciudad de México
1,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,59.0,2026-04-29,WAL-MART EXPRESS,WALMART EXPRESS SUCURSAL GIRASOLES,"RANCHO SAN LORENZO 6, ESQ. CALZADA DE LAS BOMB...",GIRASOLES III,04920,Coyoacán,Ciudad de México
2,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,62.0,2026-04-29,MERCADO PUBLICO,MERCADO MIGUEL HIDALGO,"BLVD. SANCHEZ TABOADA 9351, ENTRE CALZ. INDEPE...",ZONA RIO TIJUANA,22010,Tijuana,Baja California
3,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,62.0,2026-04-29,MERCADO PUBLICO,MERCADO MIGUEL HIDALGO,"BLVD. SANCHEZ TABOADA 9351, ENTRE CALZ. INDEPE...",ZONA RIO TIJUANA,22010,Tijuana,Baja California
4,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,54.0,2026-04-29,CHEDRAUI,CHEDRAUI SUCURSAL LEÓN POLIFORUM,"BLVD. FRANCISCO VILLA 105, ESQ. BLVD. ADOLFO L...",EL TLACOACHE,37500,León,Guanajuato
5,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,54.0,2026-04-29,CHEDRAUI,CHEDRAUI SUCURSAL LEÓN POLIFORUM,"BLVD. FRANCISCO VILLA 105, ESQ. BLVD. ADOLFO L...",EL TLACOACHE,37500,León,Guanajuato
6,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,42.8,2026-04-29,MEGA SORIANA,MEGA SORIANA SUCURSAL COAPA,"PROLONGACION DIVISION DEL NORTE1555, ENTRE ESC...",CHIMAL,14700,Tlalpan,Ciudad de México
7,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,42.8,2026-04-29,MEGA SORIANA,MEGA SORIANA SUCURSAL COAPA,"PROLONGACION DIVISION DEL NORTE1555, ENTRE ESC...",CHIMAL,14700,Tlalpan,Ciudad de México
8,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,42.8,2026-04-29,MERCADO SORIANA,MERCADO SORIANA SUCURSAL LA SIERRA,"CALZ. SILVESTRE TERRAZAS 9001, ENTRE 24 DE FEB...",RAMON REYES,31410,Chihuahua,Chihuahua
9,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,42.8,2026-04-29,MERCADO SORIANA,MERCADO SORIANA SUCURSAL LA SIERRA,"CALZ. SILVESTRE TERRAZAS 9001, ENTRE 24 DE FEB...",RAMON REYES,31410,Chihuahua,Chihuahua


In [9]:
df_profeco.describe()

,precio
count,1866.000000
mean,58.098124
std,11.095239
min,29.900000
25%,54.900000
50%,57.000000
75%,60.000000
max,100.000000


In [20]:
df_profeco.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1866 entries, 0 to 1865
Data columns (total 11 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   producto           1866 non-null   object 
 1   tipo_producto      1866 non-null   object 
 2   precio             1866 non-null   float64
 3   fecha_observacion  1866 non-null   object 
 4   cadena_comercial   1866 non-null   object 
 5   establecimiento    1866 non-null   object 
 6   direccion          1866 non-null   object 
 7   colonia            1866 non-null   object 
 8   cp                 1866 non-null   object 
 9   municipio          1866 non-null   object 
 10  entidad            1866 non-null   object 
dtypes: float64(1), object(10)
memory usage: 160.5+ KB


In [12]:
#municipios unicos que hay en el df
df_profeco['entidad'].unique()

array(['Ciudad de México', 'Baja California', 'Guanajuato', 'Chihuahua',
       'Estado de México', 'Chiapas', 'Querétaro', 'Tabasco', 'Campeche',
       'Coahuila', 'Quintana Roo', 'Veracruz', 'Jalisco', 'Tlaxcala',
       'Sonora', 'Michoacán', 'Zacatecas', 'Oaxaca', 'Aguascalientes',
       'Puebla', 'San Luis Potosí', 'Sinaloa', 'Baja California Sur',
       'Yucatán', 'Guerrero', 'Nuevo León', 'Durango', 'Morelos',
       'Tamaulipas', 'Hidalgo'], dtype=object)

In [15]:
df_cdmx = df_profeco[df_profeco['entidad'] == 'Ciudad de México']

df_cdmx.head()

#muicipios de la cdmx

df_cdmx['municipio'].unique()

array(['Coyoacán', 'Tlalpan', 'Iztapalapa', 'Benito Juárez',
       'Miguel Hidalgo', 'Iztacalco', 'Cuauhtémoc', 'Gustavo A. Madero',
       'Venustiano Carranza', 'Xochimilco', 'Azcapotzalco'], dtype=object)

No tenemos información sobre la alcaldía tlahuac que es nuestro punto de interés, asi que procedemos a tomar datos que estén en los alrededores, como xochimilco e iztapalapa

In [17]:
df_xochimilco = df_cdmx[df_cdmx['municipio'] == 'Xochimilco']
print('Registros en xochimilco:', len(df_xochimilco))

df_xochimilco.head()

Registros en xochimilco: 4


,producto,tipo_producto,precio,fecha_observacion,cadena_comercial,establecimiento,direccion,colonia,cp,municipio,entidad
460,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,55.0,2026-05-12,BODEGA AURRERA,BODEGA AURRERA SUCURSAL XOCHIMILCO,"PROL. DIV. DEL NORTE 5333, CASI ESQ. GUADALUPE I",POTRERO SAN BERNARDINO,16030,Xochimilco,Ciudad de México
747,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,46.9,2026-05-19,BODEGA AURRERA,BODEGA AURRERA SUCURSAL XOCHIMILCO,"PROL. DIV. DEL NORTE 5333, CASI ESQ. GUADALUPE I",POTRERO SAN BERNARDINO,16030,Xochimilco,Ciudad de México
1463,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,49.0,2026-06-02,BODEGA AURRERA,BODEGA AURRERA SUCURSAL XOCHIMILCO,"PROL. DIV. DEL NORTE 5333, CASI ESQ. GUADALUPE I",POTRERO SAN BERNARDINO,16030,Xochimilco,Ciudad de México
1767,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,49.0,2026-06-02,BODEGA AURRERA,BODEGA AURRERA SUCURSAL XOCHIMILCO,"PROL. DIV. DEL NORTE 5333, CASI ESQ. GUADALUPE I",POTRERO SAN BERNARDINO,16030,Xochimilco,Ciudad de México


Se tomarán datos de la Bodega Aurrera Xochimilco en PROL. DIV. DEL NORTE 5333, CASI ESQ. GUADALUPE, que está a aproximadamente 7.35 km del punto de venta de Aguacates Monarca

In [30]:
df_xochimilco['colonia'].unique()

array(['POTRERO  SAN BERNARDINO'], dtype=object)

In [18]:
df_iztapalapa = df_cdmx[df_cdmx['municipio'] == 'Iztapalapa']
print('Registros en iztapalapa:', len(df_iztapalapa))

df_iztapalapa.head()


Registros en iztapalapa: 82


,producto,tipo_producto,precio,fecha_observacion,cadena_comercial,establecimiento,direccion,colonia,cp,municipio,entidad
22,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,55.0,2026-04-29,BODEGA AURRERA,BODEGA AURRERA SUCURSAL IZTAPALAPA,"CALZ. ERMITA IZTAPALAPA 875, ENTRE TLAHUAC Y D...",SANTA ISABEL INDUSTRIAL,09820,Iztapalapa,Ciudad de México
23,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,55.0,2026-04-29,BODEGA AURRERA,BODEGA AURRERA SUCURSAL IZTAPALAPA,"CALZ. ERMITA IZTAPALAPA 875, ENTRE TLAHUAC Y D...",SANTA ISABEL INDUSTRIAL,09820,Iztapalapa,Ciudad de México
40,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,60.0,2026-04-29,MERCADO PUBLICO,MERCADO LA PURISIMA,"LUIS HIDALGO MONROY 423, ESQ. QUETZAL",LEYES DE REFORMA 1A. SECCIÓN,09340,Iztapalapa,Ciudad de México
41,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,60.0,2026-04-29,MERCADO PUBLICO,MERCADO LA PURISIMA,"LUIS HIDALGO MONROY 423, ESQ. QUETZAL",LEYES DE REFORMA 1A. SECCIÓN,09340,Iztapalapa,Ciudad de México
72,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,57.0,2026-04-29,WAL-MART,WALMART SUCURSAL PLAZA ORIENTE,"CANAL DE TEZONTLE 1520, ESQ EJE 5 OTE. ROJO GOMEZ",U.H. CUCHILLA DEL MORAL II,09319,Iztapalapa,Ciudad de México


In [23]:
df_iztapalapa['colonia'].unique()

array(['SANTA ISABEL INDUSTRIAL', 'LEYES DE REFORMA 1A. SECCIÓN',
       'U.H. CUCHILLA DEL MORAL II', 'LOS ANGELES', 'SAN JUAN XALPA',
       'EJIDOS DEL MORAL', 'UNIDAD MODELO', 'EL RETOÑO', 'SECTOR POPULAR'],
      dtype=object)

In [26]:
df_iztapalapa[df_iztapalapa['colonia'] == 'SAN JUAN XALPA']

,producto,tipo_producto,precio,fecha_observacion,cadena_comercial,establecimiento,direccion,colonia,cp,municipio,entidad
112,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,42.8,2026-04-29,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México
113,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,42.8,2026-04-29,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México
359,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,59.6,2026-05-06,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México
418,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,59.6,2026-05-06,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México
520,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,56.9,2026-05-13,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México
1592,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,48.9,2026-06-03,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México
1593,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,48.9,2026-06-03,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México
1863,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,48.9,2026-06-03,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México


La colomia que queda más cerca de los olivos es San Juan Xalpa, en especial el MERCADO SORIANA SUCURSAL SAN LORENZO XALPA que se encuentra a 3.35 km de Walmart Tlahuac, el establecimiento más cercano a el punto de venta de Aguacates Monarca.

In [24]:
df_edomex = df_profeco[df_profeco['entidad'] == 'Estado de México']

df_edomex.head()

#muicipios de la cdmx

df_edomex['municipio'].unique()


array(['Tultitlán', 'Coacalco', 'Toluca', 'Lerma', 'Tlalnepantla',
       'Ecatepec', 'Huixquilucan', 'Naucalpan', 'Cuautitlán',
       'Cuautitlán Izcalli', 'Otzolotepec', 'Metepec'], dtype=object)

Ningún municipio deisponible en EdoMex queda cerca de Los olivos, Tlahuac.

## Datos cercanos al punto de venta de Aguacates Monarca

In [32]:
colonias_deseadas = ['SAN JUAN XALPA', 'POTRERO  SAN BERNARDINO']

df_filtrado = df_profeco[df_profeco['colonia'].isin(colonias_deseadas)]

print(f'Registros para utilizar: {len(df_filtrado)}')

display(df_filtrado.head(10))

Registros para utilizar: 12


,producto,tipo_producto,precio,fecha_observacion,cadena_comercial,establecimiento,direccion,colonia,cp,municipio,entidad
112,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,42.8,2026-04-29,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México
113,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,42.8,2026-04-29,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México
359,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,59.6,2026-05-06,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México
418,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,59.6,2026-05-06,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México
460,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,55.0,2026-05-12,BODEGA AURRERA,BODEGA AURRERA SUCURSAL XOCHIMILCO,"PROL. DIV. DEL NORTE 5333, CASI ESQ. GUADALUPE I",POTRERO SAN BERNARDINO,16030,Xochimilco,Ciudad de México
520,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,56.9,2026-05-13,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México
747,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,46.9,2026-05-19,BODEGA AURRERA,BODEGA AURRERA SUCURSAL XOCHIMILCO,"PROL. DIV. DEL NORTE 5333, CASI ESQ. GUADALUPE I",POTRERO SAN BERNARDINO,16030,Xochimilco,Ciudad de México
1463,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,49.0,2026-06-02,BODEGA AURRERA,BODEGA AURRERA SUCURSAL XOCHIMILCO,"PROL. DIV. DEL NORTE 5333, CASI ESQ. GUADALUPE I",POTRERO SAN BERNARDINO,16030,Xochimilco,Ciudad de México
1592,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,48.9,2026-06-03,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México
1593,"AGUACATE, S/M, 1 KG. HASS",AGUACATE,48.9,2026-06-03,MERCADO SORIANA,MERCADO SORIANA SUCURSAL SAN LORENZO XALPA (986),"AV. SAN LORENZO 2171, ENTRE NAUTLA Y ESTRELLA",SAN JUAN XALPA,09850,Iztapalapa,Ciudad de México


In [35]:
#Guardamos en disco
df_profeco.to_csv('precios_competencia_profeco.csv', index=False, encoding='utf-8')

In [36]:
df_filtrado.to_csv('precios_competencia_profeco_delimitado.csv', index=False, encoding='utf-8')